examples/configureBuiltinPlugin.md

```diff
@@ -0,0 +1,349 @@
+
+[plugins]: ../docs/plugins
+[plugin]: ../docs/plugins
+
+[fields]: ../docs/fields.md
+[fields in a kustomization file]: ../docs/fields.md
+
+[TransformerConfig]: https://github.com/kubernetes-sigs/kustomize/blob/master/pkg/transformers/config/transformerconfig.go
+
+[kustomization]: https://github.com/kubernetes-sigs/kustomize/blob/master/docs/glossary.md#kustomization
+
+
+# Customizing kustomize
+
+The [fields in a kustomization file] allow the user to
+specify which resource files to use as input, how to
+_generate_ new resources, and how to _transform_ those
+resources - add labels, patch them, etc.
+
+These fields are simple (low argument count) directives.
+For example, the `commonAnnotations` field demands only a
+list of _name:value_ pairs.
+
+If using a field triggers behavior that pleases the user,
+everyone's happy.
+
+If not, the user can ask for new behavior to be implemented
+in kustomize proper (and wait), or the user can write a
+transformer or generator [plugin].  This latter option
+generally means writing code; a Go plugin, a Go binary,
+a C++ binary, a `bash` script - something.
+
+There's a third option.  If one merely wants to tweak
+behavior that already exists in kustomize, one may be able
+to do so by just writing some YAML.
+
+## Configure the builtin plugins
+
+All of kustomize's builtin operations are implemented 
+and usable as plugins.
+
+The mapping is:
+
+ | kustomization field | builtin plugin |
+ |-----|-----|
+ | `secretGenerator` | _SecretGenerator_ |
+ | `configMapGenerator` | _ConfigMapGenerator_ |
+ | `namespace` | _NamespaceTransformer_ |
+ | `patches`   | _PatchTransformer_ |
+ | `commonLabels` | _LabelTransformer_ |
+ | `commonAnnotations` | _AnnotationTransformer_ |
+ | `namePrefix`  | _PrefixSuffixTransformer_ |
+ | `nameSuffix`  | _PrefixSuffixTransformer_ |
+ | `image` | _ImageTagTransformer_ |
+ | `replicas` | _ReplicasTransformer_ |
```

Comment:
How about adding `patchesStrategicMerge` and `patchesJson6902` as well since we haven't deprecated them?

Reply:
Done